sunkit_image/enhance.py

```diff
@@ -0,0 +1,224 @@
+    qmed : `float`
+        The scaling factor with which the median is multiplied to fill the values below `zmin`.
+        Defaults to 1.0
+
+    Returns
+    -------
+    new_image : `numpy.ndarray`
+        Background supressed image.
+    """
+
+    zmed = np.median(image)
+
+    new_image = np.where(image < zmin, qmed * zmed, image)
+
+    return new_image
+
+
+def bandpass_filter(image, nsm1=1, nsm2=3):
+
+    """
+    Applies a band pass filter to the image.
+
+    Parameters
+    ----------
+    image : `numpy.ndarray`
+        Image to be filtered.
+    nsm1 : `int`
+        Low pass filter boxcar smoothing constant.
+        Defaults to 1.
+    nsm2 : `int`
+        High pass filter boxcar smoothing constant.
+        The value of `nsm2` equal to `nsm1 + 1` gives the best enhancement.
+        Defaults to 3.
+
+    Returns
+    -------
+    new_image : `numpy.ndarray`
+        Bandpass filtered image.
+    """
+
+    if(nsm1 >= nsm2):
+        raise ValueError("nsm1 should be less than nsm2")
+
+    return ndimage.uniform_filter(image, nsm1) - ndimage.uniform_filter(image, nsm2)
+
+
+def occult2(smap, zmin, num_loop, noise_thresh, qmed=1, nsm1=1, nsm2=3, rmin=30, nmax=1000):
+
+    """
+    Implements the OCCULT-2 algorithm for loop tracing in solar images.
+
+    Parameters
+    ----------
+    smap : `sunpy.map`
+        Map on which loops are to be detected.
+    zmin : `float`
+        The minimum value of intensity which is allowed.
+    num_loop : `int`
+        The maximum number of loops to be detected per image.
+    noise_thresh : `float`
+        The intensity value below which pixels are considered noisy.
+    qmed : `float`
+        The scaling factor with which the median is multiplied to fill the values below `zmin`.
+        Defaults to 1.0
+    nsm1 : `int`
+        Low pass filter boxcar smoothing constant.
+        Defaults to 1.
+    nsm2 : `int`
+        High pass filter boxcar smoothing constant.
+        The value of `nsm2` equal to `nsm1 + 1` gives the best enhancement.
+        Defaults to 3.
+    rmin : `int`
+        The minimum radius of curvature of the loop to be detected.
+        Defaults to 30.
+    nmax : `int`
+        Maximum number of loops to be detected.
+        Defaults to 1000.
+
+    Returns
+    -------
+    new_image : `sunpy.map`
+        Image with loops marked.
+
+    References
+    ----------
+    * Markus J. Aschwanden, Bart De Pontieu, Eugene A. Katrukha.
+      Optimization of Curvi-Linear Tracing Applied to Solar Physics and Biophysics.
+    """
+
+    # 1. Supress the background
+    image = background_supression(smap.data, zmin, qmed)
+
+    # 2. Bandpass filter
+    image = bandpass_filter(image, nsm1, nsm2)
+
+    x, y = np.meshgrid(*[np.arange(v.value) for v in smap.dimensions]) * u.pix
+    coords = smap.pixel_to_world(x, y).transform_to(frames.Helioprojective)
+
+
+    # Creating the three starting arrays
+    n_s = 100
```

Comment:
It would be good if you could provide references to the equations, or paragraphs in the paper for what these variables are.  Also, variable names can be much more descriptive.  There is no need to stick to what is in the IDL code or in the paper.  It would be much better to have descriptive names so that later users can understand what is going on.

Reply:
Okay I have tried to use the exact variable names as mentioned in the paper. Will see to add more verbosity.